quisp/rules/RuleSet.h

```diff
@@ -53,7 +53,8 @@ class RuleSet : public std::list<pRule> {
   void addRule(Rule* r) { push_back(pRule(r)); };
   void addRule(pRule& r) { push_back(pRule(std::move(r))); };
   void setRule_ptr(Rule* r_ptr) { this->entangled_partner_rule.push_back(r_ptr); };
-  std::vector<Rule*> getRule_ptr() { return this->entangled_partner_rule; };
+  Rule* getRule_ptr(int i) { return this->entangled_partner_rule.at(i); };
+  int checkRule_ptrSize() { return this->entangled_partner_rule.size(); };
```

Comment:
this seems misleading name. this returns a number of entangled_partner rules, and this method checks nothing.
this name sounds like it checks the size of the pointer and returns 0 or 1 to notice it is valid or not.

how about `numRules` or `numEntangledPartnerRules` ?

Reply:
Right. Let me use `numRules`.